examples/app.py

```diff
@@ -130,44 +142,41 @@ def records():
     )
     with open(data_path) as source:
         with db.session.begin_nested():
-            for data in split_blob(source.read()):
-                Record.create(marc21.do(create_record(data)))
-
+            for index, data in enumerate(split_blob(source.read()), start=1):
+                # create uuid
+                rec_uuid = uuid.uuid4()
+                # create pid
+                rec_pid = str(index)
+                # create PID
+                PersistentIdentifier.create(
+                    'recid', rec_pid, object_type='rec', object_uuid=rec_uuid,
+                    status=PIDStatus.REGISTERED
+                )
+                # do translate
+                record = marc21.do(create_record(data))
+                # update with control number
+                record.update({'control_number': rec_pid})
```

Comment:
you should be able to use `minter`(https://github.com/inveniosoftware/invenio-pidstore/blob/master/invenio_pidstore/minters.py#L32)